qt/scientific_interfaces/Indirect/IndirectFitAnalysisTab.cpp

```diff
@@ -460,267 +860,566 @@ void IndirectFitAnalysisTab::updatePlot(
+  ensureAppendCompatibility(inputWS, guessWorkspace);
+  const auto spectrum = selectedSpectrum();
+
+  auto extractAlg =
+      extractSpectraAlgorithm(inputWS, spectrum, spectrum, startX(), endX());
+  extractAlg->execute();
+  MatrixWorkspace_sptr extracted = extractAlg->getProperty("OutputWorkspace");
+
+  auto appendAlg = appendSpectraAlgorithm(extracted, guessWorkspace);
+  appendAlg->execute();
+  MatrixWorkspace_sptr inputAndGuess =
+      appendAlg->getProperty("OutputWorkspace");
+
+  auto axis = Mantid::Kernel::make_unique<TextAxis>(2);
+  axis->setLabel(0, "Sample");
+  axis->setLabel(1, "Guess");
+  inputAndGuess->replaceAxis(1, axis.release());
+  return inputAndGuess;
 }
 
-/*
- * Adds properties to the specified property for each parameter of the
- * function with the same name as the name of the specified property.
- * Properties have names in the format: FunctionName.ParameterName
- * Adds default values for the parameters of the function, into this fit
- * analysis tab's double manager.
- *
- * @param functionGroup The group property to add parameters to.
- * @param addParameters Adds parameters from function to QtProperty if True,
- *                      else returns empty QtProperty.
- */
-QtProperty *
-IndirectFitAnalysisTab::createFunctionProperty(QtProperty *functionGroup,
-                                               const bool &addParameters) {
-  const auto &functionName = functionGroup->propertyName();
-  const auto parameters = getFunctionParameters(functionName);
-
-  for (const auto &parameter : parameters) {
-    QString paramName = functionName + "." + parameter;
-    m_properties[paramName] = m_dblManager->addProperty(parameter);
-    m_dblManager->setDecimals(m_properties[paramName], NUM_DECIMALS);
-
-    if (addParameters)
-      functionGroup->addSubProperty(m_properties[paramName]);
-  }
-  return functionGroup;
+/**
+ * Ensures one workspace is able to be appended to another.
+ *
+ * @param inputWS   The workspace to maintain compatibility with.
+ * @param spectraWS The workspace to make compatible.
+ */
+void IndirectFitAnalysisTab::ensureAppendCompatibility(
+    MatrixWorkspace_sptr inputWS, MatrixWorkspace_sptr spectraWS) const {
+  spectraWS->setInstrument(inputWS->getInstrument());
+  spectraWS->replaceAxis(0, inputWS->getAxis(0)->clone(spectraWS.get()));
+  spectraWS->setDistribution(inputWS->isDistribution());
+}
+
+/**
+ * Creates an algorithm for extracting the spectra from a specified workspace.
+ *
+ * @param inputWS     The workspace to extract spectra from.
+ * @param startIndex  The index of the first spectrum to be retained.
+ * @param endIndex    The index of the last spectrum to be retained.
+ * @param startX      The x-value that is within the first bin to be retained.
+ * @param endX        The x-value that is within the last bin to be retained.
+ * @return            An algorithm for extracting spectra.
+ */
+IAlgorithm_sptr IndirectFitAnalysisTab::extractSpectraAlgorithm(
```

Comment:
The name suggests that this method extracts `SpectraAlgorithm`.
The caller of this method actually needs the result of this algorithm. Why cannot this method execute the algorithm and return the output workspace?